### ISSUE TITLE
gradient accumulation 

### ISSUE THREAD

**Issue by tlscabinet** [KevinMusgrave/pytorch-metric-learning#327]
Hello, I am searching for the way to implement gradient accumulation, can you suggest any solution?

**Comment by KevinMusgrave** [comment]
This might work (I haven't tested it):

```python
from pytorch_metric_learning.trainers import MetricLossOnly

class TrainerGradientAccumulation(MetricLossOnly):
    def __init__(self, *args, accumulation_steps=10, **kwargs):
        super().__init__(*args, **kwargs)
        self.accumulation_steps = accumulation_steps

    def forward_and_backward(self):
        self.zero_losses()
        self.update_loss_weights()
        self.calculate_loss(self.get_batch())
        self.loss_tracker.update(self.loss_weights)
        self.backward()
        self.clip_gradients()
        if (self.iteration + 1) % self.accumulation_steps == 0:
            self.zero_grad()
            self.step_optimizers()
```
